extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java

```diff
@@ -136,6 +136,18 @@ private String guessDriver(Optional<JdbcDataSourceBuildItem> jdbcDataSource) {
                     .add(new ReflectiveClassBuildItem(true, false, QuarkusQuartzConnectionPoolProvider.class.getName()));
         }
 
+        config.triggerListeners.values().forEach((value) -> {
+            reflectiveClasses.add(new ReflectiveClassBuildItem(true, false, value.clazz));
+        });
+
+        config.jobListeners.values().forEach((value) -> {
+            reflectiveClasses.add(new ReflectiveClassBuildItem(true, false, value.clazz));
+        });
+
+        config.plugins.values().forEach((value) -> {
+            reflectiveClasses.add(new ReflectiveClassBuildItem(true, false, value.clazz));
+        });
```

Comment:
I think we should also add validation here that the `class` is a class implementing a `org.quartz.JobListener` interface. This way, we would throw the error earlier in build time. Let me know if you'd need help in doing it.

/cc @mkouba

Reply:
See https://github.com/quarkusio/quarkus/issues/10962#issuecomment-664231324

Reply:
Thanks for the feedback! I did the changes you asked.
For the validation, the only thing I was not sure of which the exception I should throw. I decided on the IllegalArgumentException. Is it ok?

Reply:
This is fine for me. 

I am thinking maybe we can have a unit tests for validation. The [UnsupportedClusteredJobConfigurationTest.java](https://github.com/quarkusio/quarkus/blob/master/extensions/quartz/deployment/src/test/java/io/quarkus/quartz/test/UnsupportedClusteredJobConfigurationTest.java) can be of inspiration.  

What do you think?

Reply:
Nice idea! And done!